### PR TITLE
Adds button to hamburger menu to format code

### PR DESCRIPTION
**Pull request by adrianfalleiro**
This PR adds a button to the hamburger menu to format code.

### CODE REVIEW

**Comment by arrowtype** [comment]
Wow, I’m glad I asked the community for someone who knows React & Redux. This was more involved than I realized! I can’t comment on the exact code quality here (hopefully @outoftime can take a look soon), but thanks so much for hopping in and making it happen.

File: src/components/__tests__/HamburgerMenu.test.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+describe('hamburger menu', () => {
+  let menu;
+
+  beforeEach(() => {
+    menu = renderComponent();
+  });
+
+  it('calls the passed `onAutoFormat()` method when format code button is pressed', () => {
+    const firstMenuItem = menu.root.findAllByType(MenuItem)[0];
```
Comment (outoftime) [inline src/components/__tests__/HamburgerMenu.test.jsx:63]: Would it be fairly straightforward to scan for a menu item with the text label we expect? I could see this turning into a false failure if a future change introduced a menu item above format code that was always on, for instance.

**Review by outoftime — APPROVED** [review]
Left one comment on the tests, but not a blocker—this looks great, thanks! Reviewing from my phone, will merge once I can get in front of a computer to do a quick manual validation.

File: src/components/__tests__/HamburgerMenu.test.jsx (hunk 1 above)
Comment (adrianfalleiro) [reply src/components/__tests__/HamburgerMenu.test.jsx:63]: @outoftime This is a good callout. The issue is that the translation function `i18next.t()` is mocked and returns a fixed value. 

What I can do is amend the mock to return back the translation key and use that to select the menu item.

Ideally would be good to use `data-testid` attrs in combination with `@testing-library/react` or `enzyme` but that's for another PR.

**Pull request merged**